components/List/src/MDCSelfSizingStereoCell.m

```diff
@@ -0,0 +1,464 @@
+  BOOL shouldVerticallyCenterTitleText = hasOnlyTitleText && hasLeadingImage;
+  if (shouldVerticallyCenterTitleText) {
+    CGFloat leadingImageViewCenterY = CGRectGetMidY(self.leadingImageViewFrame);
+    CGFloat textContainerCenterY = CGRectGetMidY(self.textContainerFrame);
+    CGFloat difference = textContainerCenterY - leadingImageViewCenterY;
+    CGRect offsetTextContainerRect = CGRectOffset(self.textContainerFrame, 0, -difference);
+    BOOL willExtendPastMargin = offsetTextContainerRect.origin.y < kVerticalMarginMax;
+    if (!willExtendPastMargin) {
+      self.textContainerFrame = offsetTextContainerRect;
+    }
+  }
+}
+
+- (CGFloat)calculateHeight {
+  CGFloat maxHeight = 0;
+  CGFloat leadingImageViewRequiredVerticalSpace = 0;
+  CGFloat trailingImageViewRequiredVerticalSpace = 0;
+  CGFloat textContainerRequiredVerticalSpace = 0;
+  if (!CGRectEqualToRect(self.leadingImageViewFrame, CGRectZero)) {
+    leadingImageViewRequiredVerticalSpace = CGRectGetMaxY(self.leadingImageViewFrame) +
+    [self verticalMarginForImageViewOfSize:self.leadingImageViewFrame.size];
+    if (leadingImageViewRequiredVerticalSpace > maxHeight) {
+      maxHeight = leadingImageViewRequiredVerticalSpace;
+    }
+  }
+  if (!CGRectEqualToRect(self.trailingImageViewFrame, CGRectZero)) {
+    trailingImageViewRequiredVerticalSpace = CGRectGetMaxY(self.trailingImageViewFrame) +
+    [self verticalMarginForImageViewOfSize:self.trailingImageViewFrame.size];
+    if (trailingImageViewRequiredVerticalSpace > maxHeight) {
+      maxHeight = trailingImageViewRequiredVerticalSpace;
+    }
+  }
+  if (!CGRectEqualToRect(self.textContainerFrame, CGRectZero)) {
+    textContainerRequiredVerticalSpace = CGRectGetMaxY(self.textContainerFrame) +
+    kVerticalMarginMax;
+    if (textContainerRequiredVerticalSpace > maxHeight) {
+      maxHeight = textContainerRequiredVerticalSpace;
+    }
+  }
+  CGFloat calculatedHeight = (CGFloat)ceil((double)maxHeight);
+  return calculatedHeight;
+}
+
+- (CGSize)sizeForImage:(UIImage *)image {
+  CGSize maxSize = CGSizeMake(kImageSideLengthMax, kImageSideLengthMax);
+  if (!image || image.size.width <= 0 || image.size.height <= 0) {
+    return CGSizeZero;
+  } else if (image.size.width > maxSize.width || image.size.height > maxSize.height) {
+    CGFloat aspectWidth = maxSize.width / image.size.width;
+    CGFloat aspectHeight = maxSize.height / image.size.height;
+    CGFloat aspectRatio = MIN(aspectWidth, aspectHeight);
+    return CGSizeMake(image.size.width * aspectRatio, image.size.height * aspectRatio);
+  } else {
+    return image.size;
+  }
+}
+
+- (CGFloat)verticalMarginForImageViewOfSize:(CGSize)size {
+  CGFloat leadingImageHeight = size.height;
+  if (leadingImageHeight > 0 && leadingImageHeight <= kImageSideLengthMedium) {
+    return kVerticalMarginMax;
+  } else {
+    return kVerticalMarginMin;
+  }
+}
+
+- (CGFloat)dynamicInterLabelVerticalPaddingWithTitleLabel:(UILabel *)titleLabel
```

Comment:
InterLabel? Is this the spacing between the title and detail labels? Can we make this naming simpler?

Reply:
Yes, it's the spacing between the labels. Is there a name you'd prefer?

Reply:
`- (CGFloat)labelsVerticalSpacing` ?